atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraClientPoolImpl.java

```diff
@@ -245,29 +248,49 @@ private CassandraClientPoolImpl(
     private void tryInitialize() {
         cassandra.cacheInitialCassandraHosts();
 
-        refreshPoolFuture = refreshDaemon.scheduleWithFixedDelay(
-                () -> {
-                    try {
-                        refreshPool();
-                    } catch (Throwable t) {
-                        log.warn(
-                                "Failed to refresh Cassandra KVS pool."
-                                        + " Extended periods of being unable to refresh will cause perf degradation.",
-                                t);
-                    }
-                },
-                config.poolRefreshIntervalSeconds(),
-                config.poolRefreshIntervalSeconds(),
-                TimeUnit.SECONDS);
-
         // for testability, mock/spy are bad at mockability of things called in constructors
         if (startupChecks == StartupChecks.RUN) {
             runOneTimeStartupChecks();
         }
         refreshPool(); // ensure we've initialized before returning
+        scheduleNextRefresh(0);
         metrics.registerAggregateMetrics(blacklist::size);
     }
 
+    private void runAndScheduleNextRefresh(int attempt) {
+        try {
+            refreshPool();
+        } catch (Throwable t) {
+            log.warn(
+                    "Failed to refresh Cassandra KVS pool."
+                            + " Extended periods of being unable to refresh will cause perf degradation.",
+                    t);
+        }
+
+        scheduleNextRefresh(attempt);
+    }
+
+    // TODO(mdaudali): We can inline this into runAndScheduleNextRefresh once #6753 is merged
+    private void scheduleNextRefresh(int attempt) {
+        if (getCurrentPools().isEmpty()) {
+            int maxShift = Math.min(MAX_ATTEMPTS_BEFORE_CAPPING_BACKOFF, attempt);
+
+            // Caps out at 2^7 * 1000 = 64000
+            int millisTillNextRefresh = Math.max(1, ThreadLocalRandom.current().nextInt((1 << maxShift) * 1000));
```

Comment:
min clamp at 1 is mostly to not require rewriting a bunch of tests that verify a single refresh, which gets violated if you generate 0 (this happens on the first load, + random luck which happened more often when the state space was smaller (seconds, not millis)